docs/source/python/integration/python_r.rst

```diff
@@ -0,0 +1,315 @@
+
+.. code-block:: bash
+
+    $ pip install rpy2
+
+The most basic thing we can do with our ``addthree`` function is to
+invoke it from Python with a number and see how it will return the result.
+
+To do so we can create an ``addthree.py`` file which uses ``rpy2`` to
+import the ``addthree`` function from ``addthree.R`` file and invoke it:
+
+.. code-block:: python
+
+    import rpy2.robjects as robjects
+
+    # Load the addthree.R file
+    r_source = robjects.r["source"]
+    r_source("addthree.R")
+
+    # Get a reference to the addthree function
+    addthree = robjects.r["addthree"]
+
+    # Invoke the function
+    r = addthree(3)
+
+    # Access the returned value
+    value = r[0]
+    print(value)
+
+Running the ``addthree.py`` file will show how our Python code is able
+to access the ``R`` function and print the expected result:
+
+.. code-block:: bash
+
+    $ python addthree.py 
+    6.0
+
+If instead of passing around basic data types we want to pass around
+Arrow Arrays, we can do so relying on the
+`rpy2-arrow <https://rpy2.github.io/rpy2-arrow/version/main/html/index.html>`_ 
+module which implements ``rpy2`` support for Arrow types.
+
+``rpy2-arrow`` can be installed through ``pip``:
+
+.. code-block:: bash
+
+    $ pip install rpy2-arrow
+
+``rpy2-arrow`` implements converters from PyArrow objects to R Arrow objects,
+this is done without occurring into any data copy cost as it relies on the
+C Data interface.
+
+To pass to ``addthree`` a PyArrow array our ``addthree.py`` needs to be modified
+to enable ``rpy2-arrow`` converters and then pass the PyArrow array:
+
+.. code-block:: python
+
+    import rpy2.robjects as robjects
+    from rpy2_arrow.pyarrow_rarrow import (rarrow_to_py_array,
+                                           converter as arrowconverter)
+    from rpy2.robjects.conversion import localconverter
+
+    r_source = robjects.r["source"]
+    r_source("addthree.R")
+
+    addthree = robjects.r["addthree"]
+
+    import pyarrow
+
+    array = pyarrow.array((1, 2, 3))
+
+    # Enable rpy2-arrow converter so that R can receive the array.
+    with localconverter(arrowconverter):
+        r_result = addthree(array)
+
+    # The result of the R function will be an R Environment
+    # we can convert back the Environment to a pyarrow Array
+    # using the rarrow_to_py_array function
+    py_result = rarrow_to_py_array(r_result)
+    print("RESULT", type(py_result), py_result)
+
+Running the newly modified ``addthree.py`` should now properly execute
+the R function and print the resulting PyArrow Array:
+
+.. code-block:: bash
+
+    $ python addthree.py
+    RESULT <class 'pyarrow.lib.DoubleArray'> [
+      4,
+      5,
+      6
+    ]
+
+.. note::
+
+    Even though we sent an ``Int64Array`` to R, we end up with a
+    result as a ``DoubleArray``. That's due to the lack of native
+    support for 64 bits numbers in R and thus its use of doubles
+    to represent those.
```

Comment:
If the script were instead (note `3L` versus `3`) I _think_ we would get an integer back.

```
    library(arrow)

    addthree <- function(arr) {
        return(arr + 3L)
    }
```

Reply:
Good catch! I actually forgot that R goes for numerics by default